doc: PR and Issues templates + contributing guidelines (#271)
SysBioChalmers/GECKO: .github/CONTRIBUTING.md

When creating the issue, please make sure:

* You tested your code (if any) with all of GECKO's [requirements](https://github.com/SysBioChalmers/GECKO).
* You did your analysis in the `main` branch of the repository.
* You provide any necessary files/links needed for understanding the issue.
* You checked that a similar issue does not exist already

> Feel free to also comment on any of the [open issues](https://github.com/SysBioChalmers/GECKO/issues). When doing so, please comply with our [code of conduct](https://github.com/SysBioChalmers/GECKO/blob/zzzzzzz/.github/CODE_OF_CONDUCT.md).

Reviewer: ```suggestion
Feel free to also comment on any of the [open issues](https://github.com/SysBioChalmers/GECKO/issues). When doing so, please comply with our [code of conduct](https://github.com/SysBioChalmers/GECKO/blob/main/.github/CODE_OF_CONDUCT.md).
```

Note that steps 3, 4, 5 and 7 can be done, if you prefer, with any git client, such as [Github Desktop](https://desktop.github.com/).

Try to keep pull requests relatively small: they should ideally provide one new feature or fix one bug. A few small bug-fixes maybe combined, but avoid packing pull requests with too many diverse and somewhat unrelated development. Smaller pull requests reduce reviewing time and make sure that code updates will make it into new releases faster.

> Finally, and for larger features that you want to work on collaboratively, you may consider to first request to join our development team to get write access to the repository so that you can create a branch directly in the main repository (or simply ask the administrator to create a branch for you). Once you have a new branch, you can push your changes directly to the main repository and when finished, submit a pull request from that branch to `devel`. [See below](#development-team-guidelines) for more details.

Reviewer: ```suggestion
Finally, and for larger features that you want to work on collaboratively, you may consider to first request to join our development team to get write access to the repository so that you can create a branch directly in the main repository. On this new branch, you can push your changes directly to the main repository and when finished, submit a pull request from that branch to `devel`. [See below](#development-team-guidelines) for more details.
```

Every pull request must be approved by at least one reviewer before it can be merged. When reviewing someone else's pull request, keep in mind the following aspects:

> * **Functionality:** The automated tests that are run as GitHub Actions should pass without any problems. Have you written a new function? Make sure that it is also tested in `tests/unittests/

Reviewer: ```suggestion
* **Functionality:** The automated tests that are run as GitHub Actions should pass without any problems. Have you written a new function? Make sure that it is also tested in `tests/unittests/`.
```

Author: ```suggestion
* **Functionality:** The automated tests that are run as GitHub Actions should pass without any problems. Have you written a new function? Make sure that it is also tested in `tests/unit_tests/geckoCoreFunctionTests.m`.
```

When releasing, please follow these steps:

1. Create a pull request from `devel` to `main`:
   - Specify the intended version in the title, e.g. `GECKO 3.0.1`
   - Indicating all new features/fixes/etc. and referencing every previous pull request included (examples [here](https://github.com/SysBioChalmers/GECKO/releases)).
   - If any [issue](https://github.com/SysBioChalmers/GECKO/issues) gets solved in the release, write in the pull request description "Closes #X", where "X" is the issue number. That way the issue will be automatically closed after merge.

> 2. Wait at least a day + 1 approval + passing GitHub Actions.

Reviewer: ```suggestion
2. Wait at least a day for at least one  approval. The GitHub Actions must also pass successfully.
```
